### ISSUE TITLE
new margin system doesn't handle negative values in bars

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#7175]
While working on https://github.com/matplotlib/matplotlib/pull/5538#issuecomment-206662711 (allow artists to opt-out from being considered by the autoscaler), I noticed that the new margins machinery doesn't handle negative values in bar plots very well:

```
from matplotlib import pyplot as plt
fig, axs = plt.subplots(1, 2)

# wanted behavior: remove the margin at y=0.
axs[0].bar(0, 1)

# unwanted behavior: remove the margin at y=-1
axs[1].bar(0, -1)

plt.show()
```

![bar](https://cloud.githubusercontent.com/assets/1322974/18817039/fc0df558-830a-11e6-8d2b-b1e77b5250a5.png)

attn @mdboom

**Comment by tacaswell** [comment]
attn @mdboom

**Comment by efiring** [comment]
closed by #7476
